### PR TITLE
Fixes possible null pointer dereference.

### PR DESCRIPTION
**Pull request by valeriyvan**
Solves static analyser issue.

### CODE REVIEW

File: src/doc.cpp [hunk 1]
```diff
@@ -1538,16 +1538,19 @@ int Doc::PrepareLyricsEnd(FunctorParams *functorParams)
 {
     PrepareLyricsParams *params = dynamic_cast<PrepareLyricsParams *>(functorParams);
     assert(params);
-
-    if ((params->m_currentSyl && params->m_lastNote) && (params->m_currentSyl->GetStart() != params->m_lastNote)) {
-        params->m_currentSyl->SetEnd(params->m_lastNote);
+    auto params_m_currentSyl = params->m_currentSyl;
```
Comment (lpugin) [inline src/doc.cpp]: This line is useless, isn't it?

File: src/doc.cpp [hunk 2]
```diff
@@ -1538,16 +1538,19 @@ int Doc::PrepareLyricsEnd(FunctorParams *functorParams)
...
+    if (!params_m_currentSyl) {
+        return FUNCTOR_STOP; // early return
+    }
+    if (params->m_lastNote && params_m_currentSyl->GetStart() != params->m_lastNote) {
+        params_m_currentSyl->SetEnd(params->m_lastNote);
     }
     else if (m_options->m_openControlEvents.GetValue()) {
-        if ((params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_i)
-            || (params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_m)) {
+        auto wordpos = params_m_currentSyl->GetWordpos();
```
Comment (lpugin) [inline src/doc.cpp]: Please change auto to the actual type

File: src/doc.cpp (hunk 1 above)
Comment (lpugin) [reply src/doc.cpp]: I meant: you do not need another local variable

Comment (valeriyvan) [reply src/doc.cpp]: No, we don't need extra variable.
But. 
1) Pointer dereference isn't free. If it's used twice it's not guaranteed compiler will optimise this. But it's not the point.
2) Point is it simplifies code reading and understanding.

File: src/doc.cpp (hunk 2 above)
Comment (valeriyvan) [reply src/doc.cpp]: Hm. Ok, I will do that. Type inference is sweet feature!

File: src/doc.cpp (hunk 1 above)
Comment (lpugin) [reply src/doc.cpp]: I find it more confusing. Having a check and early return is OK but without the extra variable.

**Comment by valeriyvan** [comment]
Look how it is now, please.
Is it ok?
Or would you like to get rid of intermediate variable `params_m_currentSyl` at all?

**Comment by lpugin** [comment]
Please remove the params_m_currentSyl variable. Otherwise it looks good

**Comment by valeriyvan** [comment]
done

File: src/doc.cpp [hunk 3]
```diff
@@ -1538,13 +1538,15 @@ int Doc::PrepareLyricsEnd(FunctorParams *functorParams)
...
+        return FUNCTOR_STOP; // early return
+    }
+    if (params->m_lastNote && params->m_currentSyl->GetStart() != params->m_lastNote) {
         params->m_currentSyl->SetEnd(params->m_lastNote);
     }
     else if (m_options->m_openControlEvents.GetValue()) {
-        if ((params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_i)
-            || (params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_m)) {
+        sylLog_WORDPOS wordpos = params->m_currentSyl->GetWordpos();
+        if (wordpos == sylLog_WORDPOS_i || wordpos == sylLog_WORDPOS_m) {
```
Comment (lpugin) [inline src/doc.cpp]: Sorry, one last thing I am noticing: please put both tests in ()

Comment (valeriyvan) [reply src/doc.cpp]: https://en.cppreference.com/w/c/language/operator_precedence
Why are these extra parentheses needed?

Comment (lpugin) [reply src/doc.cpp]: Are you offering me a C++ course? ;-)
It is a habit and prefer to have the code consistent. Thanks

Comment (valeriyvan) [reply src/doc.cpp]: Well, your game - your rules.
As for me, parentheses which don't change natural operations precedence just create visual garbage. They might be useful in complex conditions. But rare in one-liners.
Pushed change.

Comment (craigsapp) [reply src/doc.cpp]: > As for me, parentheses which don't change natural operations precedence just create visual garbage.

Absolutely not.

Comment (valeriyvan) [reply src/doc.cpp]: Ok. Different people - different opinions.
I have returned parentheses back.

**Pull request merged**